jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java

```diff
@@ -49,22 +49,24 @@ public PostResponse updatePostByUserNameAndTitle(PostRequest postRequest, String
                 .map(post -> {
                     log.debug("Updating post with title '{}' for user '{}'", title, userName);
                     this.postMapper.updateReferenceValues(postRequest, post);
-                    Post updatedPost = this.postRepository.save(post);
-                    return this.postMapper.postToPostResponse(updatedPost);
+                    return this.postRepository.save(post);
                 })
+                .map(this.postMapper::postToPostResponse)
```

Comment:
@coderabbitai, please help me to execute this line of code outside of transaction

Reply:
@coderabbitai, still `this.postMapper.postToPostResponse(updatedPost)` is under the method which is annotated with `@Transactional` so db connection is not released till this is executed

Reply:
@coderabbitai, annotating private method with @Tranasactional wont work.

Reply:
@coderabbitai, above code is same as creating a new transaction at method level, perform everything and then release the transaction when done. Instead of this, can we facade pattern so the fetch will be done in readonly Transaction, next two steps of updating and saving will be done under transaction and mapping will be out of transaction.

Reply:
@coderabbitai, please create an issue to implement the same